src/serena/config/serena_config.py

```diff
@@ -143,18 +151,40 @@ def autogenerate(
                         f"  project_name: {project_name}\n"
                         f"  language: python  # or typescript, java, csharp, rust, go, ruby, cpp, php, swift, elixir, terraform, bash\n"
                     )
-                # find the language with the highest percentage
-                dominant_language = max(language_composition.keys(), key=lambda lang: language_composition[lang])
-                languages_to_use = [dominant_language]
+                # sort languages by number of files found
+                languages_and_percentages = sorted(language_composition.items(), key=lambda item: item[1], reverse=True)
+                # find the language with the highest percentage and enable it
+                top_language_pair = languages_and_percentages[0]
+                other_language_pairs = languages_and_percentages[1:]
+                languages_to_use = [top_language_pair[0]]
+                # if in interactive mode, ask the user which other languages to enable
+                if len(other_language_pairs) > 0 and interactive:
+                    print(
+                        "Detected and enabled main language '%s' (%.2f%% of source files)." % (top_language_pair[0], top_language_pair[1])
+                    )
+                    print("Additionally detected %s other languages." % (len(other_language_pairs)))
+                    print("\nNote: Enable only languages you need symbolic retrieval/editing capabilities for.")
+                    print("      Additional language servers use resources and some languages may require additional")
+                    print("      system-level installations/configuration (see Serena documentation).")
+                    print("\nWhich additional languages do you want to enable?")
+                    for lang, perc in other_language_pairs:
```

Comment:
it will be annoying to have to answer "no" many times if many languages are detected. I propose to make a while loop where the user can write a language shorthand to add a language or write "done" to exit the loop and proceed. We can add `get_shorthand` to the `Language` enum.

Note that the user may use this to create a project in an empty directory, where no language has been detected. If no language has been detected, we could show all languages as candidates.

Reply:
I have updated the exception text.